cmd/capdctl/main.go

```diff
@@ -293,11 +294,31 @@ func apply(cfg *rest.Config, objs []runtime.Object) error {
 func printAll(objs []runtime.Object) error {
 	// Stolen from https://github.com/kubernetes/kubernetes/blob/664edf832777cb7d6d00d38ccbcd4acba1497dc1/staging/src/k8s.io/kubectl/pkg/scheme/scheme.go#L37
 	encoder := unstructured.JSONFallbackEncoder{Encoder: scheme.Codecs.LegacyCodec(scheme.Scheme.PrioritizedVersionsAllGroups()...)}
-
-	for _, obj := range objs {
+	namespaces, remainingObjects := extractNamespaces(objs)
+	for _, ns := range namespaces {
+		if err := encoder.Encode(ns, os.Stdout); err != nil {
+			return errors.WithStack(err)
+		}
+	}
+	for _, obj := range remainingObjects {
 		if err := encoder.Encode(obj, os.Stdout); err != nil {
 			return errors.WithStack(err)
 		}
 	}
 	return nil
 }
+
+func extractNamespaces(objects []runtime.Object) ([]runtime.Object, []runtime.Object) {
```

Comment:
i think Kustomize has a feature that does this but since i'm using it as a library i haven't figured out how to take advantage of it yet.

Reply:
How do you mean?

Reply:
If you use kustomize the command line tool it should sort namespaces first on output objects.

Reply:
I will probably want to do something like this:

https://github.com/kubernetes-sigs/kustomize/blob/master/plugin/builtin/inventorytransformer/InventoryTransformer.go#L100